File: PennywizeServer/Models/Transaction.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace PennywizeServer.Models
{
    public class Transaction
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }

        public DateTime Date { get; set; }
        public double Amount { get; set; }
        public string Type { get; set; }
        public string Description { get; set; }

        public string UserId { get; set; }
        public User User { get; set; }
    }
}

File: pennywize-client/src/app/models/transaction.ts
export class Transaction {
  id: string;
  date: Date;
  amount: number;
  type: string;
  description: string;

  constructor(partial?: Partial<Transaction>) {
    Object.assign(this, partial);
    this.date = this.date && new Date(this.date);
  }
}
